proposals/lip-Introduce_cross-chain_update_mechanism.md

```diff
@@ -0,0 +1,620 @@
+*   If `inboxUpdate` contains a non-empty `messageWitness`, then update `newInboxRoot` to the output of `calculateRootFromRightWitness(newInboxSize, newInboxAppendPath, inboxUpdate.messageWitness.siblingHashes)` as specified in [LIP 0031](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0031.md).
+*   Then validate the newly updated root against the certificate state root using the provided `outboxRootWitness`. 
+    Using notation from [LIP 0039][SMT-LIP], this is done via the function `verify(queryKeys, proof, certificate.stateRoot)` with 
+    ```java
+    queryKeys = [outboxKey],
+    
+    proof = {siblingHashes: outboxRootWitness.siblingHashes, 
+             queries: [{ key: outboxKey, 
+                         value: newInboxRoot, 
+                         bitmap: outboxRootWitness.bitmap }],
+            }
+            
+    outboxKey = MODULE_ID_INTEROPERABILITY 
+                || STORE_PREFIX_OUTBOX 
+                || SHA-256(uint32be(sendingChainID)).
+    ``` 
+
+If `params` contains an empty `certificate` and a non-empty `inboxUpdate`, the validity of the latter is checked as follows:
+
+*   Let `newInboxRoot`, `newInboxAppendPath` and `newInboxSize` be the resulting new root, append path and size of the Merkle tree that would be obtained if all messages from `crossChainMessages` would be appended to the `sendingAccount` inbox. Note that `sendingAccount.inbox` is not updated here.
+*   If `inboxUpdate` contains a `messageWitness`, then update `newInboxRoot` to the output of `calculateRootFromRightWitness(newInboxSize, newInboxAppendPath, inboxUpdate.messageWitness.siblingHashes)` as specified in [LIP 0031](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0031.md).
+*   Then validate that `newInboxRoot == sendingAccount.partnerChainOutboxRoot`.
+
+
+### Execute Cross-chain Updates
+
+Cross-chain updates posted on the mainchain or sidechains follow a different execution logic, which we describe in the following two subsections, and a common one, described in the ["Common Processing" section](#common-processing). 
+
+In the following, let `CCU` be the cross-chain update transaction. Let `partnerChainID = CCU.sendingChainID` and `partnerChain` the entry in the interoperability store with store prefix `STORE_PREFIX_ACCOUNT` and store key `uint32be(partnerChainID)`.
+
+
+#### Cross-chain Updates Posted on Mainchain
+
+Cross-chain update transactions posted on mainchain are transactions with 
+
+*   `moduleID = MODULE_ID_INTEROPERABILITY`,
+*   `commandID  = COMMAND_ID_MAINCHAIN_CCU`. 
+
+Then, the following is done in the given order:
+
+*   If `partnerChain.status == CHAIN_REGISTERED`, set `partnerChain.status = CHAIN_ACTIVE`
+*   For every `CCM` in `inboxUpdate.crossChainMessages`:
+    *   Validate that `params.sendingChainID == CCM.sendingChainID`.
+    *   Validate the format of `CCM` according to the function provided in [LIP "Introduce cross-chain messages"][CCM-LIP]. 
```

Comment:
I think it's worth mentioning the function name and link to the function directly.
```suggestion
    *   Validate the format of `CCM` according to the `validateFormat` function provided in [LIP "Introduce cross-chain messages"](https://research.lisk.com/t/introduce-cross-chain-messages/299#validateformat-27). 
```

Reply:
The same in the next section

Reply:
changed